### ISSUE TITLE
Deliver from GitLab.com

### ISSUE THREAD

**Issue by GCSBOSS** [jsdelivr/jsdelivr#18058]
GitLab.com is nowadays a lot more populous with public projects and it would be nice to deliver from it (just like we do from GitHub). In principle, this support must be really close to how GitHub is approached. 

Thanks for your effort!

**Comment by MartinKolarik** [comment]
Hi @GCSBOSS, sorry for a late response.

I can definitely see how this would be helpful, but unfortunately it isn't as easy as it seems - adding it to the CDN is one thing, but we'd also have to add it to [the website](https://www.jsdelivr.com/), [API](https://github.com/jsdelivr/data.jsdelivr.com), and our log parser service, which gathers stats data for the API.

Additionally, even the current support for GitHub was only meant as a backup solution for projects that are not on npm - npm allows us to provide better UX and stability than GitHub, and we advocate using it whenever possible.

In theory, we could add a custom proxy for gitlab.com, similarly to [WordPress](https://github.com/jsdelivr/jsdelivr#wordpress). That would be an easy change on our end, but proxy endpoints don't support any special features, such as version ranges, minification, directory listings or usage statistics. If we did this, I'm sure people would come here asking for those features sooner or later.

To sum it up, my recommendation would be to publish to npm if you use GitLab.

@jimaek thoughts?

**Comment by jimaek** [comment]
I dont think its that important for us, there are many things missing from jsDelivr where we need to focus.
So unless GitLab wants to sponsor the development of this feature I don't think we will do this any time soon

**Comment by brentybh** [comment]
JSDelivr serves public/open source stuffs so GitHub is enough and better than GitLab

**Comment by wuniversales** [comment]
Well, the truth is, I would like gitlab to support, even partially ...

**Comment by jimaek** [comment]
Do you have any examples of popular repos that could benefit from this integration? It would take a lot of dev resources for us to support it and it might not make sense if nobody is going to use it. Especially since both gitlab and github users push to NPM which we do support

**Comment by benatkin** [comment]
Support for importing directly from the browser and other runtimes such as Deno is getting better, and having them pre-bundled by npm is more often not what we want. Check https://deno.land/x - they consider tags to be immutable. Would be great if jsDelivr supported this with gitlab.
